src/MongoDB.Driver/MongoUrl.cs

```diff
@@ -660,19 +660,21 @@ public WriteConcern GetWriteConcern(bool enabledDefault)
         /// Resolves a connection string. If the connection string indicates more information is available
         /// in the DNS system, it will acquire that information as well.
         /// </summary>
+        /// <param name="cancellationToken">The cancellation token.</param>
         /// <returns>A resolved MongoURL.</returns>
-        public MongoUrl Resolve()
+        public MongoUrl Resolve(CancellationToken cancellationToken = default(CancellationToken))
```

Comment:
Note: this is a binary breaking change, but not a compile time breaking change.

Reply:
I guess we don't worry a lot about binary breaking changes. So, from my point of view we can leave it as is now